Fixes #35083 - add power status timeout param

Ron-Lavi opened the pull request:
When checking for the host's power status, the request stops after 3 seconds by default,
Therefore in cases where it takes longer for the host to respond, the timeout fails the request.

discussion: https://community.theforeman.org/t/bmc-timeout-setting-in-foreman-ui/29016


<!---

Thank you for contributing to The Foreman, please read the
[following guide](https://www.theforeman.org/contribute.html), in short:

* [Create an issue](https://projects.theforeman.org/projects/foreman/issues)
* Reference the issue via `Fixes #1234` in the commit message
* Prefer present-tense, imperative-style commit messages
* Mark all strings for translation, see [1]
* Suggest prerequisites for testing and testing scenarios following example above.
* Prepend `[WIP]` for work in progress to prevent bots from triggering actions
* Be patient, we will do our best to take a look as soon as we can
* Explain the purpose of the PR, attach screenshots if applicable
* List all prerequisites for testing (e.g. VMware cluster, two smart proxies...)
* Reviewers often use extensive list of items to check, have a look prior submitting [2]
* Be nice and respectful

1: https://projects.theforeman.org/projects/foreman/wiki/Translating#Translating-for-developers
2: https://github.com/theforeman/foreman/blob/develop/developer_docs/pr_review.asciidoc
-->

ofedoren reviewed (approved):
Thanks, @Ron-Lavi! LGTM, makes sense and :green_apple:

ofedoren commented on app/controllers/api/v2/hosts_controller.rb:
```diff
@@ -270,9 +270,10 @@ def power
 
       def power_status
-        render json: PowerManager::PowerStatus.new(host: @host).power_state
+        render json: PowerManager::PowerStatus.new(host: @host).power_state(params[:timeout])
```
Unless `timeout` was specified in params, this will return `nil` and pass it to the `power_state` method explicitly. This will override the default `timeout = DEFAULT_TIMEOUT` variable :/

I guess it would be safer to rewrite it a bit:
```suggestion
        render json: PowerManager::PowerStatus.new(host: @host).power_state(params[:timeout] || DEFAULT_TIMEOUT)
```

Ron-Lavi replied:
Nice catch @ofedoren !
I added a condition in the ping method itself, it should work as expected now.

ofedoren reviewed (changes requested):
Actually, I have one concern...

ofedoren commented:
@Ron-Lavi, test failures seem to be related.

Ron-Lavi commented:
@ofedoren tests are green :)

ronlavi2412 committed:
Fixes #35083 - add power status timeout param

When checking for the host's power status, the request stops after 3 seconds by default,
Therefore in cases where it takes longer for the host to respond, the timeout fails the request.

discussion: https://community.theforeman.org/t/bmc-timeout-setting-in-foreman-ui/29016

ofedoren reviewed (approved):
Thanks, @Ron-Lavi, merging :)

ofedoren commented:
CP 3.3: https://github.com/theforeman/foreman/pull/9282
